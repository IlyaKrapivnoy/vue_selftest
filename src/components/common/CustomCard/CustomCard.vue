<template>
  <el-card class="box-card my-3">
    <div class="flex justify-between items-center">
      <p class="font-bold">
        {{ props.cardTextBold }}
        <span class="font-extralight text-indigo-600">{{
          props.cardTextLight
        }}</span>
      </p>
      <div>
        <el-button
          v-for="(button, index) in props.buttons"
          :key="index"
          :type="button.type"
          plain
          @click="button.click"
        >
          {{ button.name }}
        </el-button>
      </div>
    </div>
  </el-card>
</template>

<script setup>
const props = defineProps({
  cardTextBold: String,
  cardTextLight: String,
  buttons: Array,
});
</script>
