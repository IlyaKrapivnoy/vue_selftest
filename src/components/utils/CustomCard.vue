<template>
  <el-card class="box-card my-3">
    <div class="flex justify-between items-center">
      <p class="font-bold">
        {{ props.cardTextBold }}
        <span class="font-extralight text-indigo-600">{{
          props.cardTextLight
        }}</span>
      </p>
      <div>
        <el-button
          v-for="(button, index) in props.buttons"
          :key="index"
          @click="button.click"
        >
          {{ button.name }}
        </el-button>
      </div>
    </div>
  </el-card>
</template>

<script setup>
import { defineProps } from "vue";

const props = defineProps({
  cardTextBold: String,
  cardTextLight: String,
  buttons: Array,
});
</script>
