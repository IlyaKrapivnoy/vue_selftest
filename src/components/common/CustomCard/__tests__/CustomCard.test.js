import { shallowMount } from "@vue/test-utils";
import CustomCard from "@/components/common/CustomCard/CustomCard.vue";

describe("CustomCard", () => {
  it("renders correctly", () => {
    const cardTextBold = "Bold Text";
    const cardTextLight = "Light Text";
    const buttons = [
      { type: "primary", name: "Button 1", click: jest.fn() },
      { type: "success", name: "Button 2", click: jest.fn() },
    ];

    const wrapper = shallowMount(CustomCard, {
      props: { cardTextBold, cardTextLight, buttons },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
