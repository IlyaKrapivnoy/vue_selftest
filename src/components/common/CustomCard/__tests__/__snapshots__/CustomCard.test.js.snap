// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard renders correctly 1`] = `
"<el-card class="box-card my-3">
  <div class="flex justify-between items-center">
    <p class="font-bold">Bold Text <span class="font-extralight text-indigo-600">Light Text</span></p>
    <div>
      <el-button type="primary" plain="">Button 1</el-button>
      <el-button type="success" plain="">Button 2</el-button>
    </div>
  </div>
</el-card>"
`;
